File: app/java/src/main/java/com/example/production/domain/model/process/CompletionInspectionResult.java
package com.example.production.domain.model.process;

import com.example.production.domain.model.purchase.Defect;
import lombok.*;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CompletionInspectionResult {
    private Integer id;
    private String completionResultNumber;
    private String defectCode;
    private BigDecimal quantity;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    // リレーション
    private CompletionResult completionResult;
    private Defect defect;
}

File: app/csharp/src/ProductionManagement.Domain/Models/Process/CompletionInspectionResult.cs
namespace ProductionManagement.Domain.Models.Process;

/// <summary>
/// 完成検査結果データ
/// </summary>
public class CompletionInspectionResult
{
    public int Id { get; set; }
    public required string CompletionResultNumber { get; set; }
    public required string DefectCode { get; set; }
    public required decimal Quantity { get; set; }
    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }
}
